src/v/storage/disk_log_impl.h

```diff
@@ -172,6 +195,14 @@ class disk_log_impl final : public log {
     friend class disk_log_builder;  // for tests
     friend std::ostream& operator<<(std::ostream& o, const disk_log_impl& d);
 
+    /// Compute file offset of the batch inside the segment
```

Comment:
From what I'm reading index_entry is just a hint? Shall we make it an optional rather than using {file_pos=0,offset=segment.base_offset} when we don't have it? get_file_offset already has the segment so it can compute base offest itself (if needed) when index_entry is not present


```
    } else {
        // Scan from the beginning of the segment.
        left_entry.filepos = 0;
        left_entry.offset = segments.front()->offsets().base_offset;
        vlog(
          stlog.debug,
          "Scanning (left) log segment {} from the start",
          segments.front()->offsets());
    }
    auto left_scan_bytes = co_await get_file_offset(
      segments.front(),
      left_entry,
      first,
      model::boundary_type::exclusive,
      io_priority);
```

Reply:
It would also simplify the caller. You can get rid completely of the left_entry variable